src/main/kotlin/app/revanced/patches/all/screenshot/noblock/patch/NoScreenshotBlockPatch.kt

```diff
@@ -0,0 +1,111 @@
+package app.revanced.patches.all.screenshot.noblock.patch
+
+import app.revanced.extensions.findMutableMethodOf
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.replaceInstruction
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod
+import org.jf.dexlib2.Opcode
+import org.jf.dexlib2.iface.instruction.formats.Instruction35c
+import org.jf.dexlib2.iface.reference.MethodReference
+
+@Patch(false)
+@Name("no-screenshot-block")
+@Description("Disables screenshot blocking.")
+@Version("0.0.1")
+class NoScreenshotBlockPatch : BytecodePatch() {
+
+    private companion object {
+        const val integrationClassDefinition = "Lapp/revanced/all/screenshot/noblock/NoScreenshotBlockPatch;"
+    }
+
+    // Information about method calls we want to replace
+    private enum class MethodCall(
+        val definedClassName: String,
+        val methodName: String,
+        val replacementMethodDefinition: String
+    ) {
+        SetFlags(
+            "Landroid/view/Window;",
+            "setFlags",
+            "setFlags(Landroid/view/Window;II)V",
+        );
+
+        fun replaceInstruction(method: MutableMethod, instruction: Instruction35c, instructionIndex: Int) {
+            when (this) {
+                SetFlags -> {
+                    method.replaceInstruction(
+                        instructionIndex,
+                        "invoke-static { v${instruction.registerC}, v${instruction.registerD}, v${instruction.registerE} }, ${integrationClassDefinition}->${replacementMethodDefinition}"
+                    )
+                }
+            }
+        }
+
+        companion object {
+            fun fromMethodReference(methodReference: MethodReference): MethodCall? {
+                for (type in values()) {
+                    if (type.definedClassName == methodReference.definingClass && type.methodName == methodReference.name) {
+                        return type
+                    }
+                }
```

Comment:
values().[firstOrNull](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/first-or-null.html) can be used here.

Reply:
everything done